csi/controller.go

```diff
@@ -610,6 +610,14 @@ func (s *OsdCsiServer) CreateVolume(
 		if err != nil {
 			return nil, err
 		}
+
+		_, err = volumes.Update(ctx, &api.SdkVolumeUpdateRequest{
+			VolumeId: cloneResp.GetVolumeId(),
+			Labels:   getClonedPVCMetadata(locator),
+		})
+		if err != nil {
+			return nil, err
```

Comment:
is there any need to cleanup the cloned volume before returning an error from here?

Reply:
yeah, will add this

Reply:
no longer an issue since we're handling at the sdk layer